app/src/lib/stores/app-store.ts

```diff
@@ -4167,6 +4180,21 @@ export class AppStore extends TypedBaseStore<IAppState> {
     return Promise.resolve()
   }
 
+  public _setHideWhitespaceInDiff(
+    hideWhitespaceInDiff: boolean,
+    repository: Repository,
+    file: CommittedFileChange | null
+  ): Promise<void> {
+    setBoolean(hideWhitespaceInDiffKey, hideWhitespaceInDiff)
+    this.hideWhitespaceInDiff = hideWhitespaceInDiff
+
+    if (file === null) {
+      return this.updateChangesDiffForCurrentSelection(repository)
```

Comment:
Looks like `AppStore.updateChangesDiffForCurrentSelection` was renamed to `updateChangesWorkingDirectoryDiff` in #7313, but that might not be the entire work required to update here.